tests/test_encode_decode.py

```diff
@@ -1,12 +1,62 @@
+import base64
 import quopri
 import unittest
 
 from hypothesis import given, strategies as st
 
 
 class TestBase64(unittest.TestCase):
-    # TODO: https://docs.python.org/3/library/base64.html
-    pass
+    @given(payload=st.binary())
+    def test_b64_encode_decode_round_trip(self, payload):
+        x = base64.b64encode(payload)
+        self.assertEqual(payload, base64.b64decode(x))
+
+    @given(payload=st.binary())
+    def test_standard_b64_encode_decode_round_trip(self, payload):
+        x = base64.standard_b64encode(payload)
+        self.assertEqual(payload, base64.standard_b64decode(x))
+
+    @given(payload=st.binary())
+    def test_urlsafe_b64_encode_decode_round_trip(self, payload):
+        x = base64.urlsafe_b64encode(payload)
+        self.assertEqual(payload, base64.urlsafe_b64decode(x))
+
+    @given(payload=st.binary())
+    def test_b32_encode_decode_round_trip(self, payload):
+        x = base64.b32encode(payload)
+        self.assertEqual(payload, base64.b32decode(x))
+
+    @given(payload=st.binary())
+    def test_b16_encode_decode_round_trip(self, payload):
+        x = base64.b16encode(payload)
+        self.assertEqual(payload, base64.b16decode(x))
+
+    @given(
+        payload=st.binary(),
+        foldspaces=st.booleans(),
+        wrapcol=st.integers(0, 10),
+        adobe=st.booleans(),
+    )
+    def test_a85_encode_decode_round_trip(self, payload, foldspaces, wrapcol, adobe):
+        x = base64.a85encode(
+            payload, foldspaces=foldspaces, wrapcol=wrapcol, adobe=adobe
+        )
+        self.assertEqual(
+            payload, base64.a85decode(x, foldspaces=foldspaces, adobe=adobe)
+        )
+
+    @given(payload=st.binary())
+    def test_b85_encode_decode_round_trip(self, payload):
+        x = base64.b85encode(payload)
+        self.assertEqual(payload, base64.b85decode(x))
+
+    # this is failing for payload = b'\x00'
+    # expected to remove padding implicitly
+    @unittest.expectedFailure
+    @given(payload=st.binary())
+    def test_b85_encode_with_padding_decode_round_trip(self, payload):
+        x = base64.b85encode(payload, True)
```

Comment:
Let's stick to keyword arguments - `b85encode(payload, pad=True)` is easier to interpret without checking the documentation!

What's happening here is that if the input has trailing null bytes *and is then padded with null bytes*, the trailing nulls in the message are removed along with the padding.

I'd suggest doing some experiments to see how this works (e.g. is it affected by whether the length is a multiple of four?  etc) and then using a `.filter(lambda payload: not payload.endswith(b"\x00"))` or whatever so that we just avoid trying cases which we know don't work.

Finally, please write up some detailed comments explaining what's happening and why we use such a complicated strategy!

Reply:
- Added the keyword pad. 
- I did some experiments, and the issue doesn't occur when I make length a multiple of 4. It turns out, when length%4 != 0, it adds padding with `b"\x00"`, to make it a multiple of 4.